From 0f77144b260580469ca6dfcfc11aa11b28dce038 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:17:31 -0400
Subject: [net] netfilter: nf_tables: relax string validation of NFTA_CHAIN_TYPE

Message-id: <20141021161731.22888.44043.stgit@localhost.localdomain>
Patchwork-id: 97938
O-Subject: [RHEL7 net PATCH 041/124] netfilter: nf_tables: relax string validation of NFTA_CHAIN_TYPE
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: 4c1f7818e4002ceb0498ef48cb7efa7a9597d210

 commit 4c1f7818e4002ceb0498ef48cb7efa7a9597d210
 Author: Pablo Neira <pablo@netfilter.org>
 Date:   Mon Mar 31 17:43:47 2014 +0200

    netfilter: nf_tables: relax string validation of NFTA_CHAIN_TYPE

    Use NLA_STRING for consistency with other string attributes in
    nf_tables.

    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netfilter/nf_tables_api.c b/net/netfilter/nf_tables_api.c
index 3835346..d6726b1 100644
--- a/net/netfilter/nf_tables_api.c
+++ b/net/netfilter/nf_tables_api.c
@@ -541,7 +541,7 @@ static const struct nla_policy nft_chain_policy[NFTA_CHAIN_MAX + 1] = {
         .len = NFT_CHAIN_MAXNAMELEN - 1 },
  [NFTA_CHAIN_HOOK] = { .type = NLA_NESTED },
  [NFTA_CHAIN_POLICY] = { .type = NLA_U32 },
- [NFTA_CHAIN_TYPE] = { .type = NLA_NUL_STRING },
+ [NFTA_CHAIN_TYPE] = { .type = NLA_STRING },
  [NFTA_CHAIN_COUNTERS] = { .type = NLA_NESTED },
 };
 
-- 
1.7.1